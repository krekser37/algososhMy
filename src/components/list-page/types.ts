export enum OperationTypes {
    AddHead = 'addHead',
    AddTail = 'addTail',
    DeleteHead = 'deleteHead',
    DeleteTail = 'deleteTail',
    AddByIndex = 'addByIndex',
}