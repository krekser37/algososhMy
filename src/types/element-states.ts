import { LinkedListNode } from "../components/list-page/linked-list";

export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}

export interface Step<T> {
  list: LinkedListNode<T>[];
  index?: number;
  value?: T;
}