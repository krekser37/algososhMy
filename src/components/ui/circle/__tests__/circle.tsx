import renderer from "react-test-renderer";
import { ElementStates } from "../../../../types/element-states";
import { Circle } from "../circle";
const letterText='1';
const headText= 'head';
const tailText= 'tail';
const indexText = 0;

describe("test circle component", ()=>{
it("render circle without letter",()=>{
    const circle = renderer.create(<Circle/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with letter",()=>{
    const circle = renderer.create(<Circle letter={letterText}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with head",()=>{
    const circle=renderer.create(<Circle head={headText}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with react-component Ğ² head",()=>{
    const circle=renderer.create(<Circle head={<Circle/>}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with tail",()=>{
    const circle=renderer.create(<Circle tail={tailText}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with react-component Ğ² tail",()=>{
    const circle=renderer.create(<Circle tail={<Circle/>}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with index",()=>{
    const circle=renderer.create(<Circle index={indexText}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with props 'isSmall ===  true'",()=>{
    const circle=renderer.create(<Circle isSmall={true}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with state = default",()=>{
    const circle=renderer.create(<Circle state={ElementStates.Default}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with state = changing",()=>{
    const circle=renderer.create(<Circle state={ElementStates.Changing}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

it("render circle with state = modified",()=>{
    const circle=renderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
    expect(circle).toMatchSnapshot();
})

})