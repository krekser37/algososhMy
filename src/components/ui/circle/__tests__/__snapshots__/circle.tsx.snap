// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test circle component render circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component render circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`test circle component render circle with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      1
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component render circle with props 'isSmall ===  true' 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component render circle with react-component в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component render circle with react-component в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`test circle component render circle with state = changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component render circle with state = default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component render circle with state = modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component render circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`test circle component render circle without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
